app/services/ajax.js

```diff
@@ -62,24 +62,12 @@ export default Ember.Service.extend({
       options.contentType = options.contentType || 'application/json; charset=utf-8';
     }
     success = options.success || (function() {});
-    options.success = (data, status, xhr) => {
```

Comment:
I think that we misunderstood each other when we discussed this. This code is still needed until we switch calls that may have flashes in the response. So a proper fix for now would be to inject `flashes` service here and do:

```
       if (data != null ? data.flash : void 0) {
         this.get('flashes').loadFlashes(data.flash);
       }
       if (data != null) {
         delete data.flash;
       }
```

And the same should go to the error callback.